metaserver/client.go

```diff
@@ -21,6 +29,49 @@ type Client struct {
 	client *http.Client
 }
 
+func (client *Client) GetAuthToken(privateKey *rsa.PrivateKey, authType string) (string, error) {
+	challengeURL := fmt.Sprintf("http://%[1]s/auth/%[2]s?%[2]sID=1", client.addr, authType)
```

Comment:
I don't see the public key being encoded in the challenge URL. Doesn't it need to be?

Reply:
In the metaserver's case: no, because it will have a copy of the public key and can do a lookup based on ID to retrieve it. In the provider's case, maybe? I'm not sure if provider's store public keys along with ID's currently, or even if they should.

Reply:
You may be right, but when I look at the authorization handler the first thing it does is check that the request query included the provider's ID. This seems to be hardcoded as 1 in this URL.

Reply:
Ah, yup. I'll fix that tomorrow.

Reply:
Provider should have the renterid mapped to public key from contract negotiation prior to this. And I do feel like providing it in the url could cause some security issues.

Reply:
Never mind, I spaced a bit. But it might be more ideal to pull the renters public key from the metaserver instead of passing it.